File: Selenium/Project/Project2test.py

from selenium import webdriver 
from selenium.webdriver.common.by import By
import pytest 


@pytest.fixture 
def browser():
    # Initialize the Chrome WebDriver
    driver = webdriver.Chrome()
    yield driver
    # Quit the WebDriver
    driver.quit()

def test_verify_heading(browser):
        browser.get("https://alchemy.hguy.co/jobs")

        # Print the title of the page
        
        actual_heading = browser.find_element(By.CLASS_NAME,"entry-title");
        actual_heading_text = actual_heading.text

        
        # Verify the title 
        assert actual_heading_text == 'Welcome to Alchemy Jobs', f"Expected title to be 'Welcome to Alchemy Jobs', but got '{actual_heading_text}'"
        print("The Actual Heading is:", actual_heading_text)
if __name__ == '__main__':
    pytest.main()

File: Selenium/Project/Project2Test.java

package ProjectJob;


import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Project2Test {
    WebDriver driver;
    
    // Setup function
    @BeforeClass
    public void setUp() {
        // Initialize Fire fox driver
        driver = new FirefoxDriver();
        // Open the page
        driver.get("https://alchemy.hguy.co/jobs");
    }
    
    // Test function
    @Test(priority = 1)
    public void homePageTest() {
    	
    	// Locate the heading element
        WebElement headingElement = driver.findElement(By.cssSelector("h1.entry-title"));
        String heading = headingElement.getText();
        System.out.println("heading: " + heading);
        // Assert page title
        Assert.assertEquals(heading, "Welcome to Alchemy Jobs");
        
        
    }
    
    
    // Tear down function
    @AfterClass
    public void tearDown() {
        // Close the browser
        driver.quit();
    }
}
